onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -881,7 +887,27 @@ class PlannerImpl {
           if (!node_output->Exists()) continue;
           OrtValueIndex index = Index(node_output->Name());
           ProcessDef(index, node_output);
-          plan_.SetLocation(static_cast<size_t>(index), exec_provider->GetOrtDeviceByMemType(p_kernel_def->OutputMemoryType(i)));
+          OrtDevice output_device = exec_provider->GetOrtDeviceByMemType(p_kernel_def->OutputMemoryType(i));
+          // The below ifdef is due to GetConsumerNodes()
+          // Does minimal build support CPU based providers?
+#if !defined(ORT_MINIMAL_BUILD) || defined(ORT_EXTENDED_MINIMAL_BUILD)
```

Comment:
Base minimal build is CPU EP only. There was no usage of consumer nodes in that scenario so GetConsumerNodes was excluded.